crates/rust-analyzer/src/main_loop.rs

```diff
@@ -497,10 +509,13 @@ impl GlobalState {
                 let analysis = self.snapshot().analysis;
                 move |sender| {
                     sender.send(Task::PrimeCaches(PrimeCachesProgress::Begin)).unwrap();
-                    let res = analysis.prime_caches(|progress| {
-                        let report = PrimeCachesProgress::Report(progress);
-                        sender.send(Task::PrimeCaches(report)).unwrap();
-                    });
+                    let res = analysis.parallel_prime_caches(
+                        num_cpus::get_physical().try_into().unwrap_or(u8::MAX),
```

Comment:
Well the try into should only fail if the usize wants to overflow a u8. So if you somehow have 256+ cores we will only use 255...

Reply:
Ah, that's true, carry on then

Reply:
Actually, why is this using a `u8` instead of something like a `usize`, which would match `num_cpus` and `std::thread::available_parallelism`? That seems simpler and avoids having to change this when people want to use this on 256+ core systems (which a dual-socket threadripper system can already get you).

This should also probably default to the number of logical CPUs, not physical.

Reply:
I picked physical cores to avoid over-saturating the system during indexing, and to allow on *most* machines half the CPUs to continue to be available. I don't want indexing to starve the whole system's resources. 

As for u8 vs usize. I can make that change, but the most sensible worker threads here is well below the usize limit, and I figured having 256+ threads is somewhat nonsensical for most people.

Reply:
On a dual-socket system using both sockets may increase memory contention enough that it runs slower than using only a single socket, so using at most something like 32 or 64 cores may be a good default.